pkg/deploy/graph/analysis/dc.go

```diff
@@ -27,43 +29,60 @@ const (
 func FindDeploymentConfigTriggerErrors(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 	markers := []osgraph.Marker{}
 
-dc:
 	for _, uncastDcNode := range g.NodesByKind(deploygraph.DeploymentConfigNodeKind) {
-		for _, uncastIstNode := range g.PredecessorNodesByEdgeKind(uncastDcNode, deployedges.TriggersDeploymentEdgeKind) {
-			if istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode); !istNode.Found() {
-				dcNode := uncastDcNode.(*deploygraph.DeploymentConfigNode)
-
-				// The image stream for the tag of interest does not exist.
-				// TODO: Suggest `oc create imagestream` once we have that.
-				if isNode, exists := doesImageStreamExist(g, uncastIstNode); !exists {
-					markers = append(markers, osgraph.Marker{
-						Node:         uncastDcNode,
-						RelatedNodes: []graph.Node{uncastIstNode, isNode},
-
-						Severity: osgraph.ErrorSeverity,
-						Key:      MissingImageStreamErr,
-						Message: fmt.Sprintf("The image trigger for %s will have no effect because %s does not exist.",
-							f.ResourceName(dcNode), f.ResourceName(isNode)),
-					})
-					continue dc
+		dcNode := uncastDcNode.(*deploygraph.DeploymentConfigNode)
+		marker := ictMarker(g, f, dcNode)
+		if marker != nil {
+			markers = append(markers, *marker)
+		}
+	}
+
+	return markers
+}
+
+// ictMarker inspects the image change triggers for the provided deploymentconfig and returns
+// a marker in case of the following two scenarios:
+//
+// 1. The image stream pointed by the dc trigger doen not exist.
+// 2. The image stream tag pointed by the dc trigger does not exist and there is no build in
+// 	  flight that could push to the tag.
+func ictMarker(g osgraph.Graph, f osgraph.Namer, dcNode *deploygraph.DeploymentConfigNode) *osgraph.Marker {
+	for _, uncastIstNode := range g.PredecessorNodesByEdgeKind(dcNode, deployedges.TriggersDeploymentEdgeKind) {
+		if istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode); !istNode.Found() {
+			// The image stream for the tag of interest does not exist.
+			if isNode, exists := doesImageStreamExist(g, uncastIstNode); !exists {
+				return &osgraph.Marker{
+					Node:         dcNode,
+					RelatedNodes: []graph.Node{uncastIstNode, isNode},
+
+					Severity: osgraph.ErrorSeverity,
+					Key:      MissingImageStreamErr,
+					Message: fmt.Sprintf("The image trigger for %s will have no effect because %s does not exist.",
+						f.ResourceName(dcNode), f.ResourceName(isNode)),
+					// TODO: Suggest `oc create imagestream` once we have that.
 				}
+			}
+
+			if bcNode := buildedges.BuildConfigForTag(g, istNode); bcNode != nil {
+				// Avoid warning for the dc image trigger in case there is a build in flight.
+				if latestBuild := buildedges.GetLatestBuild(g, bcNode); latestBuild != nil && !buildutil.IsBuildComplete(latestBuild.Build) {
+					return nil
```

Comment:
I see - maybe we should remove any remaing uses and that constant as well -
or at least put a comment next to that constant indicating as much.

On Sunday, May 1, 2016, Michail Kargakis <notifications@github.com> wrote:

> In pkg/deploy/graph/analysis/dc.go
> <https://github.com/openshift/origin/pull/8017#discussion_r61693645>:
>
> >  				}
> > +			}
> > +
> > +			if bcNode := buildedges.BuildConfigForTag(g, istNode); bcNode != nil {
> > +				// Avoid warning for the dc image trigger in case there is a build in flight.
> > +				if latestBuild := buildedges.GetLatestBuild(g, bcNode); latestBuild != nil && !buildutil.IsBuildComplete(latestBuild.Build) {
> > +					return nil
>
> I remember adding info level markers but @smarterclayton
> <https://github.com/smarterclayton> was against them and we endep up
> removing them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8017/files/a4f84582b0fc556083c9a4e7e2dc7c74f6ab1141#r61693645>
>